File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
    	input =~ ~r/.*\?$/          		 -> "Sure."
    	input =~ ~r/^[\t\s]*$/               -> "Fine. Be that way!"
    	input =~ ~r/^([^a-z]+\!|[^a-z\d]*)$/ -> "Whoa, chill out!"
    	true             		  			 -> "Whatever."
    end
  end
end

File: go/bob/bob.go
package bob

import "strings"

const testVersion = 3

var Sayings = []struct {
	isMatch func(string) bool
	message string
}{
	{isSilent, "Fine. Be that way!"},
	{isYelling, "Whoa, chill out!"},
	{isQuestioning, "Sure."},
}

const other = "Whatever."

func isSilent(stmt string) bool {
	return stmt == ""
}

func isYelling(stmt string) bool {
	return strings.ToUpper(stmt) == stmt &&
		strings.ToLower(stmt) != stmt
}

func isQuestioning(stmt string) bool {
	return strings.HasSuffix(stmt, "?")
}

func Hey(say string) string {
	say = strings.TrimSpace(say)

	for _, saying := range Sayings {
		if saying.isMatch(say) {
			return saying.message
		}
	}
	return other
}
